tools/validators/ontology_validator/yamlformat/validator/field_lib.py

```diff
@@ -32,6 +32,8 @@
 FIELD_CHARACTER_REGEX = re.compile(
     r'^[a-z]+[a-z0-9]*(?:_[a-z]+[a-z0-9]*)*$')
 
+# Pattern to detect the alphanumeric part of a field.
+FIELD_ALPHANUMERIC_PATTERN = re.compile('_([0-9]+)')
```

Comment:
something more robust already exists for this: 
https://github.com/google/digitalbuildings/blob/master/tools/validators/ontology_validator/yamlformat/validator/entity_type_lib.py#L31
https://github.com/google/digitalbuildings/blob/master/tools/validators/ontology_validator/yamlformat/validator/entity_type_lib.py#L166

rather than updating this, I would do the field check through the type (create a <entitytype>.HasField() method.  This is more rigorous and allows you to reuse existing functionality.